### DISCUSSION TITLE
I'm unable to do indirect expand in a view collection

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#2689] [Q&A] [answered]
`v_sites` is a view collection.  `sites` and `site_resources` are base collections.

I'm able to get the indirect expand of the `site_resources(site)` part.

`/api/collections/v_sites/records?page=1&perPage=100&expand=site,site_resources(site)`

I'm able to do a normal expand from `site_resources` to `site`, but not from the view collection.

/api/collections/site_resources/records?page=1&perPage=100&filter=site %3D 'tu4vjazm266ic8z'&expand=site

Do I need a different syntax for the view collection indirect?  It shows up as `site` being a relation in the UI.

**Comment by dschissler** [comment]
![Screenshot from 2023-06-12 03-27-28](https://github.com/pocketbase/pocketbase/assets/3968727/f9cb1271-f12b-4bb6-b94f-7320d86ce099)

![Screenshot from 2023-06-12 03-29-20](https://github.com/pocketbase/pocketbase/assets/3968727/384e9088-6f43-4779-b743-ad85ff327839)

**Comment by dschissler** [comment]
I'm thinking about changing site resources to being directly a multi-select on sites for easier use, but I'd like to know why the view collection is failing to indirectly expand.

**Comment by dschissler** [comment]
> I'm thinking about changing site resources to being directly a multi-select on sites for easier use, but I'd like to know why the view collection is failing to indirectly expand.

I'm quickly backing out of that idea due to how it makes the API Filter more complicated.

**Comment by dschissler** [comment]
I'm getting by for now by simply adding it to the view SQL.

```SQL
  JSON_GROUP_ARRAY(
    JSON_OBJECT(
      'id', sr.id,
      'title', sr.title,
      'url', sr.url
    )
  ) as resources,
// ....
LEFT JOIN siteResources sr ON sr.site = s.id

```

**Comment by ganigeorgiev** [comment]
I'm not entirely sure that I understand the described issue but note that the indirect expand performs the matching based on the root/main collection id vs the referenced collection field (in your case that seems to be `site_resources(site)`).

Or in other words, you can try to change the view collection id to be the site's id, if that makes sense for your use case (eg. `select site.id as id, ...`).

**Reply by ganigeorgiev** [reply, marked as answer]
_Or alternatively, you can also apply the indirect expand on the `site` field using dot-notation:_

```js
?expand=site.site_resources(site)
```

**Reply by dschissler** [reply]
> I'm not entirely sure that I understand the described issue but note that the indirect expand performs the matching based on the root/main collection id vs the referenced collection field (in your case that seems to be `site_resources(site)`).
> 
> Or in other words, you can try to change the view collection id to be the site's id, if that makes sense for your use case (eg. `select site.id as id, ...`).

This doesn't work.  It only works when I use `?expand=site.site_resources(site)`.

**Reply by ganigeorgiev** [reply]
Yes, sorry I forgot that in addition to the id value match we also perform a check for the collection, aka. it doesn't work because `site_resources.site` is not a `relation` field pointing to the view but to the `site` collection.

**Reply by dschissler** [reply]
I don't exactly understand this, but I at least have a way to get it to work.  It would be nice not to have to descend down through site expand to get to siteResources expand, but it's not a big deal, and possibly somehow that makes more sense anyways.

The docs could definitely use another entire page on this.  Probably View collections could be one or two pages, plus another page for expansions related to that.
